### PR TITLE
Matter Switch: use parent-child for multi-switch devices

### PR DESCRIPTION
**Pull request by ctowns**
https://smartthings.atlassian.net/browse/CHAD-11858

### CODE REVIEW

File: drivers/SmartThings/matter-switch/src/init.lua [hunk 1]
```diff
@@ -48,74 +49,56 @@ local function find_default_endpoint(device, component)
...
-    else
-      component_map[string.format("switch%d", i)] = ep
+    if ep ~= main_endpoint then -- don't create a child device that maps to the main endpoint
+      local name = string.format("%s %d", device.label, current_component_number)
+      device.log.info(string.format("creating child device %s", name))
+      driver:try_create_device(
+        {
+          type = "EDGE_CHILD",
+          label = name,
+          profile = "switch-binary",
```
Comment (hubTab) [inline drivers/SmartThings/matter-switch/src/init.lua:73]: Is it possible to refer to this string by leveraging the fingerprint::deviceProfileName?

File: drivers/SmartThings/matter-switch/src/init.lua [hunk 2]
```diff
@@ -48,74 +49,56 @@ local function find_default_endpoint(device, component)
...
+          parent_device_id = device.id,
+          parent_assigned_child_key = string.format("%02X", ep),
```
Comment (hubTab) [inline drivers/SmartThings/matter-switch/src/init.lua:75]: Is there a utility function available that, provided an ep, returns its formatted value, as It is being used throughout?

**Comment by ctowns** [comment]
Running into some issues with testing and need to do some more clean up, but the plan is now to not get this in for the beta deploy today and instead target end of the week.

File: drivers/SmartThings/matter-switch/src/init.lua (hunk 1 above)
Comment (greens) [reply drivers/SmartThings/matter-switch/src/init.lua:73]: not available in the driver afaik

File: drivers/SmartThings/matter-switch/src/test/test_matter_switch_parent_child.lua [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  {
+    {
+      channel = "matter",
+      direction = "receive",
+      message = {
+        mock_device.id,
+        clusters.OnOff.attributes.OnOff:build_test_report_data(mock_device, 3, true)
+      }
+    },
+		{
```
Comment (greens) [inline drivers/SmartThings/matter-switch/src/test/test_matter_switch_parent_child.lua:152]: whitespace

**Comment by lelandblue** [comment]
What is the status of this PR is it still needed?

**Comment by ctowns** [comment]
This was merged here: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1094

**Pull request closed without merging**